extensions/liquibase-mongodb/runtime/src/main/java/io/quarkus/liquibase/mongodb/runtime/LiquibaseMongodbBuildTimeConfig.java

```diff
@@ -15,4 +15,22 @@ public class LiquibaseMongodbBuildTimeConfig {
      */
     @ConfigItem(defaultValue = "db/changeLog.xml")
     public String changeLog;
+
+    /**
+     * Flag to enable / disable the generation of the init task Kubernetes resources.
+     * This property is only relevant if the Quarkus Kubernetes/OpenShift extensions are present.
+     *
+     * The default value is `quarkus.liquibase-mongodb.enabled`.
+     */
+    @ConfigItem(defaultValue = "${quarkus.liquibase-mongodb.enabled:true}")
+    public boolean generateInitTask;
+
+    /**
+     * Flag to enable / disable the migration using the generated init task Kubernetes resources.
+     * This property is only relevant if the Quarkus Kubernetes/OpenShift extensions are present.
+     *
+     * The default value is `quarkus.liquibase-mongodb.migrate-at-start`.
+     */
+    @ConfigItem(defaultValue = "${quarkus.liquibase-mongodb.migrate-at-start:false}")
```

Comment:
I don't think we need two properties for doing pretty much the same thing.
One should be enough.

Also, I am not a big fan for having build time configuration defaulting to runtime values. This seems weird if not wrong.

Last, I am wondering if `generateInitTask` will is descriptive enough to let people understand this refers to kubernetes jobs.

Reply:
> Also, I am not a big fan for having build time configuration defaulting to runtime values. This seems weird if not wrong.
> 
I'm not super happy with this either. Though, we're generating build time resources (the init-task resources) that is actually disabled by default (`migrate-at-start` property is disabled by default), and can be enabled/disabled at runtime. 

Therefore, if users enable the `migrate-at-start` property at build time, then it means that they are likely to need the `init-task` when running the app in K8s, so I think it makes sense to me. 

About having two properties, this extension can be enabled/disabled using two properties `enabled` and `migrate-at-start`, so having only one property is probably not enough. Also, adding a totally new build-time property will be unlikely to be used by users (because they are used to use the existing and well-known `enabled` and `migrate-at-start` properties).

Reply:
The `migrate-at-start` property is a runtime property and does not affect in any way the generation of kubernetes resources, so the property is redundant and needs to be removed. What the init container does should not be a concern of the part that is generating the Job.

About the default value I don't have a super strong opinion so let's see what others think.
@geoand @gsmet: We need your expert opinion: `Is it ok to have a build time property with a default value that points to a runtime property?`

Reply:
I am pretty sure that won't work, but I'll have a look at this tomorrow

Reply:
The migration is enabled/disabled by runtime properties, and the generated Job to perform the migration has to be created/or not created at build time. So, here is the clash. 

> Is it ok to have a build time property with a default value that points to a runtime property?

It works as you can see with the new test cases I added as part of this pull request.

Reply:
> The migration is enabled/disabled by runtime properties, and the generated Job to perform the migration has to be created/or not created at build time. So, here is the clash.

Agree. There is a clash so we shouldn't tie the two together. 

> It works as you can see with the new test cases I added as part of this pull request.

The fact that this is indeed working is irrelevant. There is a clash as you already pointed out and we should better avoid it.

Reply:
> > It works as you can see with the new test cases I added as part of this pull request.
> 
> The fact that this is indeed working is irrelevant. There is a clash as you already pointed out and we should better avoid it.

I don't see your point here: the test cases probe that the new build properties whose default values are taken from runtime properties work fine, but you say it's irrelevant? 

The current pull request does address and avoids the current linked issue when using flyway / liquibase plus K8s/OpenShift extensions by using well-known properties. It's perfectly fine if you all don't like this solution and/or prefer adding a new build time property that users will need to learn to generate the init-task resources. But don't agree with you saying that the test cases are irrelevant or that the clashing problem is not addressed.